builtin/providers/aws/resource_aws_autoscaling_group.go

```diff
@@ -304,6 +312,8 @@ func resourceAwsAutoscalingGroupUpdate(d *schema.ResourceData, meta interface{})
 		AutoScalingGroupName: aws.String(d.Id()),
 	}
 
+	opts.NewInstancesProtectedFromScaleIn = aws.Bool(d.Get("protect_from_scale_in").(bool))
```

Comment:
This is indeed the correct way of updating - otherwise we will be passing an unnecessary parameter to the API each time

I was able to use the same code you had and it updated as expected.